src/sql/parts/commandLine/common/commandLineService.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import { ConnectionProfile } from 'sql/parts/connection/common/connectionProfile';
+import { ICommandLineProcessing } from 'sql/parts/commandLine/common/commandLine';
+import { IConnectionManagementService } from 'sql/parts/connection/common/connectionManagement';
+import { ICapabilitiesService } from 'sql/services/capabilities/capabilitiesService';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import * as Constants from 'sql/parts/connection/common/constants';
+import { IQueryEditorService } from 'sql/parts/query/common/queryEditorService';
+import * as platform from 'vs/platform/registry/common/platform';
+import { ConnectionProviderProperties, IConnectionProviderRegistry, Extensions as ConnectionProviderExtensions } from 'sql/workbench/parts/connection/common/connectionProviderExtension';
+import * as TaskUtilities from 'sql/workbench/common/taskUtilities';
+import { IObjectExplorerService } from 'sql/parts/objectExplorer/common/objectExplorerService';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+
+export class CommandLineService implements ICommandLineProcessing {
+	private _connectionProfile : ConnectionProfile;
+	private _showConnectionDialog: boolean;
+
+	constructor(
+		@IConnectionManagementService private _connectionManagementService : IConnectionManagementService,
+		@ICapabilitiesService private _capabilitiesService : ICapabilitiesService,
+		@IEnvironmentService private _environmentService : IEnvironmentService,
+		@IQueryEditorService private _queryEditorService : IQueryEditorService,
+		@IObjectExplorerService private _objectExplorerService : IObjectExplorerService,
+		@IEditorService private _editorService : IEditorService,
+	)
+	{
+		let profile = null;
+		if (this._environmentService && this._environmentService.args.server)
+		{
+			profile = new ConnectionProfile(_capabilitiesService, null);
+			// We want connection store to use any matching password it finds
+			profile.savePassword = true;
```

Comment:
is it the case we always want to save the password?  there are some security concerns here (particularly on Linux where we don't have an encryption library and depend only on filesystem ACLs)

Reply:
ConnectionStore only looks for saved credentials that match the profile if savePassword is true. Also, we never pass any password on the command line. So if there's no saved credential that matches the server name/user name/database name combination, this connection attempt will fail, and the user will be shown the connection dialog. At that point they have the option to check/uncheck the Remember password checkbox.